### PR TITLE
Improve and develop automate_bindings.py

### PR DESCRIPTION
**Pull request by zer0main**

### CODE REVIEW

File: tools/automate_bindings.py [hunk 1]
```diff
@@ -70,11 +82,29 @@ def getSelf(module_name):
     '''
     return frame % (module_name, module_name)
 
+def findCorrespondingKeyInDict(dictionary, full_key):
+    for key in dictionary:
+        if full_key.find(key) is not -1:
```
Comment (starius) [inline tools/automate_bindings.py]: 1. Numbers should be compared with `==`, not with `is`.
2. `if key in full_key:`

File: tools/automate_bindings.py [hunk 2]
```diff
@@ -111,17 +141,38 @@ def returnValue(return_type):
...
     else:
         builtin_type = getBuiltinType(return_type)
+        # func - function for returning values to Lua
         if builtin_type:
             func_name = TYPE_TO_LUA_FUNCS[builtin_type]
         else:
             func_name = 'luawt_toLua'
-        return RETURN_CALLS_TEMPLATE % func_name
+        # Method to convert problematic type to built-in.
+        # Empty by defualt.
```
Comment (starius) [inline tools/automate_bindings.py]: typo

File: tools/automate_bindings.py [hunk 3]
```diff
@@ -111,17 +141,38 @@ def returnValue(return_type):
...
+        convert_f = ''
+        problematic_type = findCorrespondingKeyInDict(
+            PROBLEMATIC_TYPES_CONVERSIONS,
+            return_type,
+        )
+        if problematic_type:
+            convert_f += '.'
+            convert_f += PROBLEMATIC_TYPES_CONVERSIONS[problematic_type]
+            convert_f += '()'
```
Comment (starius) [inline tools/automate_bindings.py]: Can you do it in one line and use `=` instead of `+=`?

File: tools/automate_bindings.py [hunk 4]
```diff
@@ -271,10 +322,10 @@ def writeSourceToFile(module_name, source):
...
     for parameter in parameters:
-        content = readFile(parameter["filename"])
-        writeToFile(parameter["filename"], addItem(
-            parameter["pattern"],
-            parameter["module_str"],
+        content = readFile(parameter['filename'])
+        writeToFile(parameter['filename'], addItem(
+            parameter['pattern'],
+            parameter['module_str'],
             content
```
Comment (starius) [inline tools/automate_bindings.py]: `,`

**Review by starius — CHANGES_REQUESTED** [review]

File: tools/automate_bindings.py (hunk 2 above)
Comment (zer0main) [reply tools/automate_bindings.py]: Fixed :+1:

File: tools/automate_bindings.py (hunk 4 above)
Comment (zer0main) [reply tools/automate_bindings.py]: Done :+1:

File: tools/automate_bindings.py (hunk 3 above)
Comment (zer0main) [reply tools/automate_bindings.py]: Ok, done.

File: tools/automate_bindings.py (hunk 1 above)
Comment (zer0main) [reply tools/automate_bindings.py]: Done :+1:

**Review by starius — APPROVED** [review]

**Pull request merged**